pkg/controller/deployment/util/deployment_util.go

```diff
@@ -227,7 +227,7 @@ func Revision(obj runtime.Object) (int64, error) {
 
 // SetNewReplicaSetAnnotations sets new replica set's annotations appropriately by updating its revision and
 // copying required deployment annotations to it; it returns true if replica set's annotation is changed.
-func SetNewReplicaSetAnnotations(deployment *apps.Deployment, newRS *apps.ReplicaSet, newRevision string, exists bool) bool {
+func SetNewReplicaSetAnnotations(deployment *apps.Deployment, newRS *apps.ReplicaSet, newRevision string, exists bool, revHistoryLimit int) bool {
```

Comment:
`RevisionHistoryLimit` is a field that has a different meaning. Use a different name to avoid confusion.

Reply:
How about naming the parameter revHistoryLimitInChars ?